rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java

```diff
@@ -246,8 +244,30 @@ protected void prepareUndoLog(TableRecords beforeImage, TableRecords afterImage)
         }
 
         ConnectionProxy connectionProxy = statementProxy.getConnectionProxy();
-
         TableRecords lockKeyRecords = sqlRecognizer.getSQLType() == SQLType.DELETE ? beforeImage : afterImage;
+
```

Comment:
This method is use for prepare undo log, so you put the remove duplicate lock key, that s look like unreasonable.